themes/10up-theme/includes/blocks.php

```diff
@@ -42,19 +42,49 @@ function setup() {
 }
 
 /**
- * Add in blocks that are registered in this theme
+ * Automatically registers all blocks that are located within the includes/blocks directory
  *
  * @return void
  */
 function register_theme_blocks() {
 	// Filter the plugins URL to allow us to have blocks in themes with linked assets. i.e editorScripts
 	add_filter( 'plugins_url', __NAMESPACE__ . '\filter_plugins_url', 10, 2 );
 
-	// Require custom blocks.
-	require_once TENUP_THEME_BLOCK_DIR . '/example-block/register.php';
+	// Register blocks for the active theme, for both parent and child theme,
+	// if applicable.
+	foreach ( wp_get_active_and_valid_themes() as $theme ) {
+		$block_directory_path = $theme . '/includes/blocks/';
```

Comment:
I think this is the same as `TENUP_THEME_BLOCK_DIR`

https://github.com/10up/wp-scaffold/blob/trunk/themes/10up-theme/functions.php#L15

Reply:
Yes but with the additional support of child themes. 

(which might not be super important)